Lib/test/test_complex.py

```diff
@@ -242,6 +242,43 @@ def test_mul(self):
         self.assertRaises(TypeError, operator.mul, 1j, None)
         self.assertRaises(TypeError, operator.mul, None, 1j)
 
+        self.assertComplexesAreIdentical((1e300+1j) * complex(INF, INF),
```

Comment:
Is the test `complex(INF, INF) * complex(INF, double)` relevant? if so, I'd suggest using a loop so that you can see better which cases were handled:

```py
for z, w, s in [
	(complex(INF, INF), complex(1e300, 1), complex(NAN, INF)),
	...
]:
	with self.subTest(z=z, w=w, s=s):
		self.assertComplexesAreIdentical(z * w, s)
		self.assertComplexesAreIdentical(w * z, s)
```

Reply:
Changed to use subTest().

I'm not sure (rest of tests don't use this), but maybe we should change explicit constructor calls to using inf/nan/infj/nanj constants instead (e.g. ``complex(INF, NAN)`` to ``inf+nanj``).

Reply:
I find an explicit constructor more readable. I was just worried that listing the test cases one after one may make you skip one case (for reviews it's also a bit easier).